test-coverage-enforcer/src/main/java/org/wso2/testcoverageenforcer/Maven/Jacoco/CoverageReportReader.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.wso2.testcoverageenforcer.Maven.Jacoco;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.jacoco.core.analysis.Analyzer;
+import org.jacoco.core.analysis.CoverageBuilder;
+import org.jacoco.core.analysis.IBundleCoverage;
+import org.jacoco.core.analysis.IPackageCoverage;
+import org.jacoco.core.tools.ExecFileLoader;
+import org.wso2.testcoverageenforcer.Application;
+import org.wso2.testcoverageenforcer.Constants;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Iterator;
+
+/**
+ * This class represents a jacoco coverage report with a relevant jacoco execution data file and capable of
+ * getting line coverage information
+ */
+public class CoverageReportReader {
+
+    private static final Logger log = Logger.getLogger(Application.class);
+    private final String title;
+    private final File executionDataFile;
+    private final File classesDirectory;
+    private ExecFileLoader execFileLoader;
+    private IBundleCoverage bundleCoverage;
+
+    /**
+     * Create a new reader for the jacoco report file
+     *
+     * @param projectBuildDirectory Project build folder containing classes files and jacoco exec report file
+     * @throws FileNotFoundException Cannot Find any jacoco execution file
+     */
+    public CoverageReportReader(final File projectBuildDirectory) throws IOException {
+
+        this.title = new File(projectBuildDirectory.getPath().replace(Constants.Build.BUILD_TARGET_FOLDER, Constants.EMPTY_STRING)).getName();
+        Iterator<File> executionFiles = FileUtils.iterateFiles(projectBuildDirectory, new String[]{Constants.Build.BUILD_EXECUTION_FILE}, true);
```

Comment:
Make the formatting based on the code standards and check this through the PR

Reply:
Fixed in 2bc0618a39bb056608ff116edf53bc2b3f4e8099.